src/test/java/org/fcrepo/client/FcrepoResponseTest.java

```diff
@@ -69,4 +79,103 @@ public void testResponse() throws IOException {
+     * repeatedly after the {@code FcrepoResponse} has been {@code close()}ed.
+     *
+     * @throws Exception if something exceptional happens
+     */
+    @Test
+    public void testClosableReleasesResources() throws Exception {
+        final InputStream mockBody = mock(InputStream.class);
+        final FcrepoResponse underTest = new FcrepoResponse(
+                URI.create("http://localhost/foo"), 201, "text/plain", URI.create("http://localhost/bar"), mockBody);
+
+        assertFalse("FcrepoResponse objects should not be closed until close() is invoked.", underTest.isClosed());
+
+        underTest.close();
+        assertTrue(underTest.isClosed());
+        verify(mockBody, times(1)).close();
+
+        underTest.close();
+        assertTrue(underTest.isClosed());
+        verify(mockBody, times(1)).close();
+    }
+
+    /**
+     * Demonstrates that if an {@code IOException} is thrown by {@link FcrepoResponse#close()}, <em>and</em> an
+     * exception is thrown inside of a client's {@code try} block, the {@code IOException} from the {@code close()}
+     * method is properly appended as a suppressed exception.
+     *
+     * @throws Exception if something exceptional happens
+     */
+    @Test
+    public void testClosableSuppressedExceptions() throws Exception {
+        final InputStream mockBody = mock(InputStream.class);
+        final String notSuppressedMsg = "Not suppressed.";
+        final String suppressedMsg = "Suppressed IOE";
+        doThrow(new IOException(suppressedMsg)).when(mockBody).close();
+
+        try (FcrepoResponse underTest = new FcrepoResponse(URI.create("http://localhost/foo"), 201, "text/plain",
+                URI.create("http://localhost/bar"), mockBody)) {
+            assertFalse(underTest.isClosed());
+
+            throw new RuntimeException(notSuppressedMsg);
+
+        } catch (Exception e) {
+            assertEquals(notSuppressedMsg, e.getMessage());
+            assertTrue(e.getSuppressed() != null && e.getSuppressed().length == 1);
+            assertEquals(suppressedMsg, e.getSuppressed()[0].getMessage());
+        }
+
+        verify(mockBody).close();
+    }
+
+    /**
+     * Demonstrates a successful idiomatic usage with try-with-resources
+     *
+     * @throws Exception if something exceptional happens
+     */
+    @Test
+    public void testIdiomaticInvokation() throws Exception {
+        final String content = "Hello World!";
+        final ByteArrayInputStream entityBody = new ByteArrayInputStream(content.getBytes());
+        final FcrepoClient client = mock(FcrepoClient.class);
+
+        when(client.get(any(URI.class), any(String.class), any(String.class))).thenReturn(
+                new FcrepoResponse(null, 200, null, null, entityBody));
+
+        try (FcrepoResponse res = client.get(URI.create("foo"), "", "")) {
+            assertEquals(content, IOUtils.toString(res.getBody()));
+        } catch (Exception e) {
```

Comment:
This is... quite general? Is there a specific `Exception` that is actually likely here?